concourse/pipelines/guest-package-build.yaml

```diff
@@ -1596,6 +1596,489 @@ jobs:
+        vars:
+          universe: cloud-yuck
+          repo: google-compute-engine-auto-updater
+          environment: stable
+  - task: publish-time-since-stable
+    file: guest-test-infra/concourse/tasks/publish-time-since.yaml
+    vars:
+      last_publish_date: ((.:last-stable-date))
+      package_name: "google-compute-engine-auto-updater"
+  - put: compute-image-windows-tag
+    params:
+      name: last-stable-tag/stable  # file contains only the word 'stable'
+      tag: last-stable-tag/stable
+      commitish: compute-image-windows-tag/commit_sha
+  - put: compute-image-windows-tag
+    params:
+      name: last-stable-tag/stable-today  # file contains 'stable-YYYYMMDD'
+      tag: last-stable-tag/stable-today
+      commitish: compute-image-windows-tag/commit_sha
+  on_success:
+    task: success
+    file: guest-test-infra/concourse/tasks/publish-job-result.yaml
+    vars:
+      pipeline: "guest-package-build"
+      job: "promote-windows-auto-updater-stable"
+      result_state: "success"
+      start_timestamp: ((.:start-timestamp-ms))
+  on_failure:
+    task: failure
+    file: guest-test-infra/concourse/tasks/publish-job-result.yaml
+    vars:
+      pipeline: "guest-package-build"
+      job: "promote-windows-auto-updater-stable"
+      result_state: "failure"
+      start_timestamp: ((.:start-timestamp-ms))
+
+- name: upload-windows-powershell-staging
+  plan:
+  - in_parallel:
+      steps:
+      - get: compute-image-windows-tag
+        passed: [build-compute-image-windows]
+      - get: guest-test-infra
+      - get: compute-image-windows
+  - task: generate-timestamp
+    file: guest-test-infra/concourse/tasks/generate-timestamp.yaml
+  - load_var: start-timestamp-ms
+    file: timestamp/timestamp-ms
+  - load_var: package-version
+    file: compute-image-windows-tag/version
+  - task: generate-build-id
+    file: guest-test-infra/concourse/tasks/generate-build-id.yaml
+    vars:
+      prefix: ""
+  - load_var: build-id
+    file: build-id-dir/build-id
+  - task: upload-windows-powershell-to-staging
+    file: guest-test-infra/concourse/tasks/gcloud-package-operation.yaml
+    vars:
+      package_path: compute-image-windows/google-compute-engine-powershell.((.:package-version)).0@1.goo
+      universe: cloud-yuck
+      repo: google-compute-engine-powershell
+    params:
+      TYPE: uploadToStaging
+  on_success:
+    task: success
+    file: guest-test-infra/concourse/tasks/publish-job-result.yaml
+    vars:
+      pipeline: "guest-package-build"
+      job: "upload-windows-powershell-staging"
+      result_state: "success"
+      start_timestamp: ((.:start-timestamp-ms))
+  on_failure:
+    task: failure
+    file: guest-test-infra/concourse/tasks/publish-job-result.yaml
+    vars:
+      pipeline: "guest-package-build"
+      job: "upload-windows-powershell-staging"
+      result_state: "failure"
+      start_timestamp: ((.:start-timestamp-ms))
+
+- name: promote-windows-powershell-stable
```

Comment:
you've added the word windows to all jobs, just a note the original job naming scheme was package-specific and tasks were release-specific. also since powershell is windows only it is a bit redundant :)

Reply:
PowerShell on Linux has pretty good support for our distros - https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-linux?view=powershell-7.2

I added it since all the packages are only windows even though names like certgen and auto-updater were very generic.